core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalCheckpointThread.java

```diff
@@ -46,6 +52,8 @@
   private final int mJournalCheckpointSleepTimeMs;
   /** Writes a new checkpoint after processing this many journal entries. */
   private final long mCheckpointPeriodEntries;
+  /** Whether we are currently creating a checkpoint. */
+  private AtomicBoolean mCheckpointing = new AtomicBoolean(false);
```

Comment:
It seems like all accesses are also synchronized on `mCheckpointing`? What are the access rules for this variable, and why does it need to be an `AtomicBoolean`?

Reply:
changed to 
```java
  /** Object for sycnhronizing accesses to mCheckpointing. */
  private final Object mCheckpointingLock = new Object();
  /** Whether we are currently creating a checkpoint. */
  @GuardedBy("mCheckpointingLock")
  private boolean mCheckpointing = false;
```

I was using `AtomicBoolean` before as a way to combine the `mCheckpointingLock` and `mCheckpointing` into one field. It's kind of awkward though since the atomicity isn't used at all, and it's confusing to see multiple types of synchronization at play.